Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2381 @@
+ do i = 1 to LPARs
+    nextEntryHandle = JSON_getArrayEntry(LPARArray,i-1)
+    LPARsList.i.uri=JSON_findValue(nextEntryHandle,"object-uri",,
+                     HWTJ_STRING_TYPE)
+    if foundJSONValue(LPARsList.i.uri) = FALSE then
+      do
+        errMsg = 'Failed to obtain uri for LPAR entry ('||i||')'
+        return fatalError(errMsg)
+      end
+
+    LPARsList.i.targetName=JSON_findValue(nextEntryHandle,"target-name",,
+                     HWTJ_STRING_TYPE)
+    if foundJSONValue(LPARsList.i.targetName) = FALSE then
+      do
+        errMsg = 'Failed to obtain target name for LPAR entry ('||i||')'
+        return fatalError(errMsg)
+      end
+
+    LPARsList.i.name=JSON_findValue(nextEntryHandle,"name",HWTJ_STRING_TYPE)
+    if foundJSONValue(LPARsList.i.name) = FALSE then
+      do
+        errMsg = 'Failed to obtain name for LPAR entry ('||i||')'
+        return fataError(errMsg)
+      end
+
+    LPARsList.i.status = JSON_findValue(nextEntryHandle,"status",,
+                     HWTJ_STRING_TYPE)
+    if foundJSONValue(LPARsList.i.status) = FALSE then
+      do
+        errMsg = 'Failed to obtain status for LPAR entry ('||i||')'
+        return fatalError(errMsg)
+      end
+
+/*  say 'index ('||i||') Processed LPAR ('||LPARsList.i.name||, */
+/*                ') target name ('||LPARsList.i.targetName||,  */
+/*                ') with uri ('||LPARsList.i.uri||,            */
+/*                ') in status ('||LPARsList.i.status').'       */
+ end /* endloop thru the JSON LPARs array */
+
+return 0
+
+/*******************************************************/
+/* Function:  QueryLPAR                                */
+/*                                                     */
+/* Retrieve LPAR attributes identified by              */
+/* LPARAttribute stem variable                         */
+/*                                                     */
+/* Return 0 if successful, otherwise return -1         */
+/*******************************************************/
+QueryLPAR:
+/* Now use the retrieved CPC uri and LPAR name
+   to retrieve crypto information for the LPAR
+
+   GET <CPC uri>/image-activation-profiles/<profile name>?properties
+*/
+
+If LPARAttribute.0 < 1 then
+  do
+    return fatalError('** no LPAR attributes to query **')
+  end
+
+LPARQueryUri = CPCuri||'/image-activation-profiles/'
+LPARQueryUri = LPARQueryUri||LPARname
+LPARQueryUri = LPARQueryuri||'?properties='
+ do i = 1 to LPARAttribute.0
+   LPARQueryUri = LPARQueryUri||LPARAttribute.i
+
+   /* more properties still to add so need a ',' */
+   if i < LPARAttribute.0 then
+     LPARQueryUri = LPARQueryUri||','
+ end
+ LPARQueryUri = LPARQueryUri||'&cached-acceptable=true'
+
+say '========================================================='
+say 'Starting to process content for LPAR '||LPARName
+
+LPARQueryResponse = GetRequest(LPARQueryUri, CPCtargetName)
+
+if LPARQueryResponse = '' Then
+  do
+    return fatalError('** Failed to retrieve Crypto info,
```

Comment:
see comment earlier in the main flow, add a new function to catch the fact the cpc is less than z16, before it ever attempts to retrieve this information

Reply:
fixed